src/feature/rend/rendservice.c

```diff
@@ -546,14 +546,48 @@ copy_service_on_prunning(rend_service_t *dst, rend_service_t *src)
   dst->n_intro_points_wanted = src->n_intro_points_wanted;
 }
 
+/** A SIGHUP occured and service pruning might have happened; some services
+ *  might not be available anymore. Check if any introduction circuits are now
+ *  useless and need to be cleaned up. */
+static void
+clean_intro_circs_after_service_prune(const smartlist_t *surviving_services) {
```

Comment:
Also, I think you can remove `service_` from the name, we are in `rendservice.c` so it is imply. I think just `clean_intro_circs_after_pruning()` will be more meaningful. No strong opinion.

Reply:
Fixed in new branch.